### PR TITLE
New resource: Add InGuest resource type software to Compute

### PR DESCRIPTION
**Pull request by abhinav-ghai**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by anuchandy — CHANGES_REQUESTED** [review]

File: specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/compute.json [hunk 1]
```diff
@@ -8209,6 +8307,56 @@
...
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/InGuestSoftwareItemProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
```
[the remark this answers is not in the document]
Comment (abhinav-ghai) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/compute.json]: Yes, it's an ARM resource. Example:
/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/guestResourceTypes/software/items/{itemId}

File: specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/compute.json [hunk 2]
```diff
@@ -2177,6 +2177,104 @@
...
+          },
+          {
+            "name": "vmName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the virtual machine."
+          },
+          {
+            "name": "itemId",
```
[the remark this answers is not in the document]
Comment (abhinav-ghai) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/compute.json]: Yes, Example:
/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/guestResourceTypes/software/items/{**itemId**}

**Comment by anuchandy** [comment]
@abhinav-gha - Larent @lmazuel noted that we are adding these apis to 2016-04-30-preview and 2018-10-01, but what about other api-versions in the middle 2017-12-01, 2018-06-01? It's weird to see that an operation exist in old API disappears in next couple of api-versions then come back in later one.

File: specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/compute.json [hunk 3]
```diff
@@ -8209,6 +8307,56 @@
...
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The uri to fetch the next page of InGuest software items on the VM. Call ListNext() with this to fetch the next page of items."
+        }
+      },
+      "required": [
+        "value"
+      ],
+      "description": "The List of Software items operation response."
```
[the remark this answers is not in the document]
Comment (abhinav-ghai) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/compute.json]: Done

File: specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/compute.json [hunk 4]
```diff
@@ -8209,6 +8307,56 @@
...
+        "version": {
+          "type": "string",
+          "description": "Specifies the version of the software."
+        },
+        "publisher": {
+          "type": "string",
+          "description": "Specifies the publisher of the software."
+        }
+      },
+      "description": "Describes the properties of an InGuest software item."
```
[the remark this answers is not in the document]
Comment (abhinav-ghai) [reply specification/compute/resource-manager/Microsoft.Compute/stable/2018-10-01/compute.json]: Done

File: specification/compute/resource-manager/readme.md [hunk 5]
```diff
@@ -177,6 +177,12 @@ directive:
     suppress:
       - DefinitionsPropertiesNamesCamelCase
 
+
+  - where:
+    - $.definitions.InGuestSoftwareItem
+    suppress:
+    - TrackedResourcePatchOperation
```
[the remark this answers is not in the document]
Comment (abhinav-ghai) [reply specification/compute/resource-manager/readme.md]: Okay

File: specification/compute/resource-manager/Microsoft.Compute/preview/2016-04-30-preview/compute.json [hunk 6]
```diff
@@ -1892,6 +1892,104 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/guestResourceTypes/software/items":{
```
Comment (ravbhatnagar) [inline specification/compute/resource-manager/Microsoft.Compute/preview/2016-04-30-preview/compute.json]: lets make the values of guestResourceTypes as an enum and add the allowed values there.

File: specification/compute/resource-manager/Microsoft.Compute/preview/2016-04-30-preview/compute.json [hunk 7]
```diff
@@ -5730,6 +5828,58 @@
...
+          "type": "string",
+          "readOnly": true,
+          "description": "The uri to fetch the next page of InGuest software items on the VM. Call ListNext() with this to fetch the next page of items."
+        }
+      },
+      "description": "The List of Software items operation response."
+    },
+    "InGuestSoftwareItemProperties": {
+      "properties": {
+        "name": {
```
Comment (ravbhatnagar) [inline specification/compute/resource-manager/Microsoft.Compute/preview/2016-04-30-preview/compute.json]: top level property names should not be repeated inside the properties bag. Lets rename this to softwareName

File: specification/compute/resource-manager/Microsoft.Compute/preview/2016-04-30-preview/compute.json [hunk 8]
```diff
@@ -1892,6 +1892,104 @@
...
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Get the InGuest software items.": {
+            "$ref": "./examples/ListInGuestSoftwareItems.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/guestResourceTypes/software/items/{itemId}":{
```
Comment (ravbhatnagar) [inline specification/compute/resource-manager/Microsoft.Compute/preview/2016-04-30-preview/compute.json]: What would GET on /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/guestResourceTypes/software return? Do you want to support this API?

**Review by ravbhatnagar — COMMENTED** [review]
Few notes, Please take a look.

**Comment by anuchandy** [comment]
@abhinav-ghai can you address ARM comments and my question on not having this feature in some api-versions?

**Comment by anuchandy** [comment]
Closing the PR to due to inactivity.

**Pull request closed without merging**